synapse/util/logcontext.py

```diff
@@ -116,8 +116,6 @@ def set_current_context(cls, context):
 
     def __enter__(self):
         """Enters this logging context into thread local storage"""
-        if self.parent_context is not None:
-            raise Exception("Attempt to enter logging context multiple times")
```

Comment:
If we're entering the context multiple times perhaps we shouldn't be tracking the context to restore on the object itself.